scalars/guide.md

```diff
@@ -1,107 +1,119 @@
-a Scalar is represented in an execution result before serialization is entirely
-up to the specific engine. In theory JSON is just one possible serialization
-format, but it is by far the most important and also the only format the GraphQL
-spec clearly defines, therefore we choose to only consider JSON here.
+A Scalar implementation can be observed in three different ways:
+
+1. the serialized result of an execution in JSON
+2. the variables JSON input
+3. the GraphQL literal input
+
+We must choose a serialization format in order to observe a Scalar
+implementation, as inside the GraphQL type system a Scalar is a primitive and
+how a Scalar is represented in an execution result before serialization is
+entirely up to the specific engine.
+
+We are choosing to only consider JSON. Although JSON is only one possible
+serialization format, it is by far the most important and the only format
+defined by the GraphQL spec.
 
 ## Coercion
 
 The process of ensuring that a value is valid and optionally converting it into
-a more appropriate representation is called "coercion". For Scalars specifically
-there are two different coercion mechanism: "Result Coercion" and "Input
-Coercion". Both are implemented as a set of functions taking an input and
-producing an output. Result coercion consists of one function and input coercion
-of up to three different ones. (Warning naming: the input of a function is not
-to be confused with "Input Coercion". Input coercion refers to the overall
-mechanism, but just "input" is what is consumed by a function.)
+a more appropriate representation is called "coercion". For Scalars, there are
+two different coercion mechanisms: "Result Coercion" and "Input Coercion". Both
+are implemented as a set of functions taking an input and producing an output.
+Result coercion consists of one function and input coercion of up to three
+different ones.
 
-## Result Coercion
+Note: The input of a function is not to be confused with "Input Coercion". The
+phrase "Input Coercion" refers to the overall mechanism, but the word "input"
+describes what is consumed by a function.
+
+## Result Coercion values and function
 
 Result coercion is implemented as a single function per Scalar type called
-"coerceResult". The input is a "raw result value" and the output a "coerced
+`coerceResult`. The input is a "raw result value" and the output a "coerced
 result value". The raw result value is the output of a previously invoked
 resolver function, which is then coerced by invoking `coerceResult`.
 
-The coerced result value is a leaf value of the overall execution result which
+The coerced result value is a leaf value of the overall execution result, which
 is serialized to JSON. The serialized JSON value can be observed and a GraphQL
 Scalar spec must clearly define the possible JSON values that a Scalar type can
 produce.
 
-## Input Coercion
+## Input Coercion values
```

Comment:
Splitting input coercion section into values and functions so it's easier to read in the spec-md layout.